### ISSUE TITLE
security group name

### ISSUE THREAD

**Issue by Dimitry1** [terraform-aws-modules/terraform-aws-rds-aurora#376]
## Is your request related to a new offering from AWS?
Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)? See [CHANGELOG.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md), too.
  - No 🛑: please wait to file a request until the functionality is avaialble in the AWS provider
  - Yes ✅: please list the AWS provider version which introduced this functionality

## Is your request related to a problem? Please describe.

would like to be able to pass in security_group_name

## Describe the solution you'd like.

pass in security_group_name not only prefix

## Additional context

would like to have well defined resources.  security_group_name is a resource that i would like to pass in as part of the security_group_creation.

**Comment by bryantbiggs** [comment]
the current form is that if creating the security group, the name follows that of the cluster - could you describe a scenario where the name of the clusters security group group should be named differently?

**Comment by Dimitry1** [comment]
thought so as well; however, what i'm observing is that cluster name becomes prefix for security group name, i.e. if i have a cluster foo, my security group name is foo-YYYYMMDDHHMMSSS (derived string)....  I would like it to either stay foo or be able to override it to be foo-bar but skip the string that lengthy date derived string

**Comment by Dimitry1** [comment]
> the current form is that if creating the security group, the name follows that of the cluster - could you describe a scenario where the name of the clusters security group group should be named differently?

Hi Bryant - wanted to quickly follow up and see if answer provided helps or not really.  It is quite consistent through the 7.x and 8.x versions.

**Comment by bryantbiggs** [comment]
have you tried setting `security_group_use_name_prefix` = `false`?

https://github.com/terraform-aws-modules/terraform-aws-rds-aurora/blob/4380f14c101b145e4b691bcb465b7978b0e44fdc/variables.tf#L549C11-L553

**Comment by Dimitry1** [comment]
> 
Brilliant - thank you!  It did the trick though i must say the way i was reading it i'd expect it to work in exact opposite way...

**Comment by estahn** [comment]
@bryantbiggs I do have a scenario where I would like the name to be different. The RDS is called after the service, e.g. `myapp-staging`. I would like the security group to be named `myapp-staging-rds` or an alternative to this. I don't mind if I can't name it myself and it is hardcoded.

Alternatively this could be provided:
```terraform
variable "security_group_name" {
  description = "Names the AWS Security Group, otherwise uses the RDS name."
  type        = string
  default     = ""
}
```

```
locals {
  security_group_name = try(coalesce(var.security_group_name, var.name))
}
```

```
resource "aws_security_group" "this" {
  ...

  name        = var.security_group_use_name_prefix ? null : local.security_group_name
  name_prefix = var.security_group_use_name_prefix ? "${local.security_group_name}-" : null
```

Would this be acceptable? I can prepare a PR if so.

**Comment by antonbabenko** [comment]
This issue has been resolved in version 8.1.0 :tada:
